src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -709,8 +705,8 @@ public void SavePackageSources(IEnumerable<PackageSource> sources, PackageSource
                         oldPackageSource,
                         existingDisabledSourceItem,
                         existingCredentialsItem,
-                        updateEnabled: sourceUpdateSettings.UpdateEnabled,
-                        updateCredentials: sourceUpdateSettings.UpdateCredentials,
+                        updateEnabled: true,
+                        updateCredentials: true,
                         shouldSkipSave: true,
                         isDirty: ref isDirty);
```

Comment:
Should the ProtocolVersion be updated as well?

Reply:
That's part of the PackageSource object on line 708.

Reply:
The method that these lines of code are part of (call site, not call target), `SavePackageSources(IEnumerable<PackageSource> sources)`, looks at each package source in the incoming list, either finds the match in the existing `nuget.config` and then call `UpdatePackageSource`, or if no match is found, call `AddPackageSource`. Finally, `SavePackageSources` will remove any sources that were in the existing nuget.config, but were not matched in the incoming `sources` list.

The lines this comment was made on are the parameters to that `UpdatePackageSource`, and the first two parameters `source` and `oldPackageSource`, are of type `PackageSource`, which has the property `ProtocolVersion`, as Nikolche pointed out.

The diff that this comment is pointing out (removing `sourceUpdateSettings` and hardcoding `true` for `updateEnabled` and `updateCredentials` is related to bundling a fix for https://github.com/NuGet/Home/issues/10015 in the same PR, to avoid confusion like https://github.com/NuGet/NuGet.Client/pull/5334#discussion_r1281096639. I didn't use git's revert feature, but it should effectively be a revert of https://github.com/NuGet/NuGet.Client/pull/3495, since the goals it tried to achieve didn't work, and actually introduced a new mode of failure (well, https://github.com/NuGet/NuGet.Client/commit/b5b44526dea0379ebd6c8e51e8a041c06d5845ca#diff-db51a3ac45fc35243298d74958a007f9073a8b044e3510325cef068b810a6d9eR101 and https://github.com/NuGet/NuGet.Client/commit/b5b44526dea0379ebd6c8e51e8a041c06d5845ca#diff-2cde950543b0ac9b52f8a0b7c8e358a3fb7804f670a12e077827c6dfcd814459R33 introduced the mode of failure, and https://github.com/NuGet/NuGet.Client/commit/86c78b646f39fbebe1349a45deb59bc1e1c60fd9 introduced the regression that this PR is fixing), which I had to fix in this PR in `NuGetSourcesService`.